pkgs/top-level/all-packages.nix

```diff
@@ -10590,6 +10590,8 @@ in
 
   knightos-z80e = callPackage ../development/tools/knightos/z80e { };
 
+  koka = haskellPackages.callPackage ../development/compilers/koka { };
```

Comment:
We should probably change this to:
```
  koka = haskell.lib.justStaticExecutables (haskellPackages.callPackage ../development/compilers/koka { });
```
so
1. We don't needlessly include all the haskell library dependencies in the closure
and
2. Users don't get a bunch of
```
error: packages '/nix/store/someProgram/lib/links/libgmpxx.4.dylib' and '/nix/store/someOtherProgram/lib/links/libgmpxx.4.dylib' have the same priority 5; use 'nix-env --set-flag priority NUMBER INSTALLED_PKGNAME' to change the priority of one of the conflicting packages (0 being the highest priority)
```
errors when trying to install it on MacOS.

Reply:
I should probably open a separate pull request for this though, since I was too late to catch this before it was merged.

Reply:
With vs without `justStaticExecutables`:
```
❯ nix path-info -Sh /nix/store/vs77d70z834c3hvzz73a87sl5xkb0w65-koka-2.1.1/
/nix/store/vs77d70z834c3hvzz73a87sl5xkb0w65-koka-2.1.1   817.0M
❯ nix path-info -Sh /nix/store/5mshznqzns5lndkv4632m886dwlvkbka-koka-2.1.1/
/nix/store/5mshznqzns5lndkv4632m886dwlvkbka-koka-2.1.1     2.4G
```

Reply:
Yeah, good point.

Reply:
I've created a pr now: https://github.com/NixOS/nixpkgs/pull/119857